src/main/java/com/ericsson/ei/services/RuleCheckService.java

```diff
@@ -0,0 +1,79 @@
+package com.ericsson.ei.services;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.ericsson.ei.handlers.EventHandler;
+import com.ericsson.ei.handlers.EventToObjectMapHandler;
+import com.ericsson.ei.jmespath.JmesPathInterface;
+import com.ericsson.ei.queryservice.ProcessAggregatedObject;
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+@Component
+public class RuleCheckService implements IRuleCheckService {
+
+    @Autowired
+    JmesPathInterface jmesPathInterface;
+
+    @Autowired
+    EventHandler eventHandler;
+
+    @Autowired
+    private ProcessAggregatedObject processAggregatedObject;
+
+    @Autowired
+    EventToObjectMapHandler eventToObjectMapHandler;
+
+    private static final Logger LOG = LoggerFactory.getLogger(RuleCheckService.class);
+
+    @Override
+    public String prepareAggregatedObject(String listRulesJson, String listEventsJson)
+            throws JSONException, JsonProcessingException, IOException {
+        JSONArray jsonObj = new JSONArray(listRulesJson);
+        JSONArray jsonObj2 = new JSONArray(listEventsJson);
+        eventHandler.getRulesHandler().setParsedJason(jsonObj.toString());
+        String aggregatedObjectId = null;
```

Comment:
aggregatedObjectId needs to be a list since the action might result in several aggregated objects.

Reply:
in our scenario (List of events & List of rules) gives only one aggregated object, ofcourse it is in List.